test/functional/ui/output_spec.lua

```diff
@@ -22,6 +22,7 @@ describe("shell command :!", function()
 
   after_each(function()
     screen:detach()
+    session.command('call jobstop(b:terminal_job_id)')
```

Comment:
Hmm, neither my original patch or removing the `after_each` actually fixes the problem.